### ISSUE TITLE
Activity Stack not properly managed

### ISSUE THREAD

**Issue by vanshikaarora** [commons-app/apps-android-commons#2345]
**On Navigating between various activities pressing back doesn't returns to home page** 

Staring from home when you visit multiple activities from navigation drawer each activity is added to the stack. On pressing back button the app visits all the activities in the stack rather than taking back to home activity. 
Is it intended to be like that?
Shouldn't the app be directed back to the default screen of navigation drawer: The HomePage

**Steps to reproduce:** 

- From home page visit explore then bookmarks then notifications then settings (or you can also follow your own sequence) 
- As you go on pressing back button the app will take you through all the activities that you have visited

_Before starting to work on this I would like to have suggestions from the members whether this Issue is relevant and sounds agreeable 
Looking forward for suggestions_

**Would you like to work on the issue?**
Yes

**Comment by vanshikaarora** [comment]
@maskaravivek @nicolas-raoul Would you like to give your suggestions?

**Comment by misaochan** [comment]
By "back button" you mean the button on the device, correct, not the "up button" in the actionbar? If so, I personally think that it is working as intended, as we already have the actionbar's up button for users to go back to home.

Am open to further debate though! :)

**Comment by vanshikaarora** [comment]
@misaochan Yes I was talking about the back button on the device. If it is intended to return to previous activity upon pressed and not the home page in case of multiple activities visited?

Also which upon button did you mention here?

**Comment by BinduVerma123** [comment]
@nicolas-raoul can i work on this issue?

**Comment by vanshikaarora** [comment]
@BinduVerma123 the issue is still under discussion. Can you also pitch in your suggestion why this feature  would be better than the previous one :)

**Comment by BinduVerma123** [comment]
@vanshikaarora adding this feature would make the app better as it will save user's time.As when the user have to reach the home page again  then he had to press the back button again and again which make the process monotonous and boring  and this may lead to lack of user's interest in the app.By adding this feature user can immediately reach to the home page and can check other features and use any other feature he/she want's without waiting for the home page to come(by repeatedly pressing the back button).

**Comment by vanshikaarora** [comment]
@misaochan @nicolas-raoul I had the same views like @BinduVerma123 while posting this issue can you please pitch in ideas or shall I close this Issue?

**Comment by misaochan** [comment]
I just took a closer look at our current version and it seems the "up" button in the action bar has been disabled. :/ I'm not sure exactly when or why that happened, it seems we need to find out. I don't think we should let it stay disabled, because allowing the user to navigate back to the home screen via Up navigation is important. https://developer.android.com/training/implementing-navigation/ancestral

This is a somewhat different issue entirely, but the point is that we should not be replicating the Up button's functionality with the back button. Google Developer's guides advise similarly: https://developer.android.com/training/design-navigation/ancestral-temporal

**Comment by nicolas-raoul** [comment]
My understanding of what would be ideal:

Let say I navigate like this
A) I am in the app's home activity
B) I am in Explore
C) I browse a category
D) I browse its sub-category
E) I browse its sub-sub-category

At the point of (E),
- If I tap the "Back" button, it should take me back to (D)
- If I tap the app icon in the upper-left, it should take me to (A)

Is that correct?

Many people enjoy [getting lost in Wikipedia](https://www.urbandictionary.com/define.php?term=Lost%20In%20The%20Wiki%27s), going from wikilink to wikilink and drifting from topic to topic. I do the same with Commons category exploration, and I often end up getting a huge stack. So I agree with @BinduVerma123 that having to "*press the back button again and again*" is frustrating when I suddenly need to check the Nearby map during a bus trip, for instance.

**Comment by misaochan** [comment]
> If I tap the "Back" button, it should take me back to (D)
> If I tap the app icon in the upper-left, it should take me to (A)

Yes, that should be the intended functionality. The Android developer guidelines recommend making this distinction between the Up button (the upper-left back arrow icon) and the Back button (in hardware). Our problem currently is that the Up button doesn't work, but that should be fixed by fixing the Up button, not replicating its functionality with the back button.

**Comment by vanshikaarora** [comment]
> Our problem currently is that the Up button doesn't work, but that should be fixed by fixing the Up button

@misaochan Should I create a separate Issue for it?

**Comment by vanshikaarora** [comment]
>So I agree with @BinduVerma123 that having to "press the back button again and again" is frustrating when I suddenly need to check the Nearby map during a bus trip, for instance.

@nicolas-raoul In your opinion should there be a back button to handle this or should we add finish to after each new intent so that the home activity appears as the homepage?
